File: 00002/prim1.py
import sys, math, random
MAXV=31630
def prime_table():
    flag = [True]*MAXV
    for i in xrange(2, MAXV/2):
        if flag:
            for j in xrange(i*2, MAXV, i):
                flag[j] = False
    return [i for i in xrange(2, MAXV) if flag[i]]

table = prime_table()
n = int(sys.stdin.readline().strip())
for c in range(n):
    x, y = sys.stdin.readline().strip().split()
    x, y = max(int(x), 2), int(y)
    flag = [True] * (y - x + 1)

    for p in table:
        if p > y: break
        low = max(x/p*p, p*2)
        for j in xrange(low, y + 1, p):
            #print p, j
            if j >= x: flag[j-x] = False

    for f in xrange(0, y - x + 1):
        if flag[f]: print f + x
    if c < n - 1: print

File: 00002/prim1.rb
maxv=31630
flag=Array.new(maxv, true)

(2..maxv/2).each do |i|
    (i*2..maxv).step(i) do |j|
        flag[j] = false
    end
end

table=Array.new

(2..maxv).each do |i|
    if flag[i]
        table.push(i)
    end
end

n = Integer(gets.chomp)

n.times do |c|
    x, y = gets.chomp.split
    x, y = [2, Integer(x)].max, Integer(y)
    flag=Array.new(y-x+1, true)

    table.each do |p|
        if p > y
            break
        end
        low = [x/p*p, p*2].max
        (low..y).step(p) do |j|
            if j >= x
                flag[j-x] = false
            end
        end
    end
    (y-x+1).times do |f|
        if flag[f]
            puts f + x
        end
    end
    if c < n - 1
        puts
    end
end
